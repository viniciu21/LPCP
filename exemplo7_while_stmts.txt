declaration:
    a: int;
    b: int;
    c: int;
end_declaration

main:
    while(11 < 10):
        a = 5;
    end_while;
end_main