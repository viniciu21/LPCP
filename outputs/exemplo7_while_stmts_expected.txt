Declaração de variável: Id "a" (2,5)(True,[(Id "a" (2,5),IntValue 0 (2,8))],[],[],[])
Declaração de variável: Id "b" (3,5)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "b" (3,5),IntValue 0 (3,8))],[],[],[])
Declaração de variável: Id "c" (4,5)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "b" (3,5),IntValue 0 (3,8)),(Id "c" (4,5),IntValue 0 (4,8))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,5)(True,[(Id "a" (2,5),IntValue 5 (8,9)),(Id "b" (3,5),IntValue 0 (3,8)),(Id "c" (4,5),IntValue 0 (4,8))],[],[],[])
Expressão:[LeftParenthesis (9,10),Id "a" (9,11),Less (9,13),IntValue 10 (9,15),RightParenthesis (9,17)] Valor: BoolValue True (8,9) Condição: True
Atualização de estado sobre a variável: Id "a" (10,9)(True,[(Id "a" (2,5),IntValue 6 (8,9)),(Id "b" (3,5),IntValue 0 (3,8)),(Id "c" (4,5),IntValue 0 (4,8))],[],[],[])
Expressão:[LeftParenthesis (9,10),Id "a" (9,11),Less (9,13),IntValue 10 (9,15),RightParenthesis (9,17)] Valor: BoolValue True (8,9) Condição: True
Atualização de estado sobre a variável: Id "a" (10,9)(True,[(Id "a" (2,5),IntValue 7 (8,9)),(Id "b" (3,5),IntValue 0 (3,8)),(Id "c" (4,5),IntValue 0 (4,8))],[],[],[])
Expressão:[LeftParenthesis (9,10),Id "a" (9,11),Less (9,13),IntValue 10 (9,15),RightParenthesis (9,17)] Valor: BoolValue True (8,9) Condição: True
Atualização de estado sobre a variável: Id "a" (10,9)(True,[(Id "a" (2,5),IntValue 8 (8,9)),(Id "b" (3,5),IntValue 0 (3,8)),(Id "c" (4,5),IntValue 0 (4,8))],[],[],[])
Expressão:[LeftParenthesis (9,10),Id "a" (9,11),Less (9,13),IntValue 10 (9,15),RightParenthesis (9,17)] Valor: BoolValue True (8,9) Condição: True
Atualização de estado sobre a variável: Id "a" (10,9)(True,[(Id "a" (2,5),IntValue 9 (8,9)),(Id "b" (3,5),IntValue 0 (3,8)),(Id "c" (4,5),IntValue 0 (4,8))],[],[],[])
Expressão:[LeftParenthesis (9,10),Id "a" (9,11),Less (9,13),IntValue 10 (9,15),RightParenthesis (9,17)] Valor: BoolValue True (8,9) Condição: True
Atualização de estado sobre a variável: Id "a" (10,9)(True,[(Id "a" (2,5),IntValue 10 (8,9)),(Id "b" (3,5),IntValue 0 (3,8)),(Id "c" (4,5),IntValue 0 (4,8))],[],[],[])
Expressão:[LeftParenthesis (9,10),Id "a" (9,11),Less (9,13),IntValue 10 (9,15),RightParenthesis (9,17)] Valor: BoolValue False (8,9) Condição: False
Tokens do programa: [Declaration (1,1),Colon (1,12),Id "a" (2,5),Colon (2,6),Type "int" (2,8),SemiColon (2,11),Id "b" (3,5),Colon (3,6),Type "int" (3,8),SemiColon (3,11),Id "c" (4,5),Colon (4,6),Type "int" (4,8),SemiColon (4,11),EndDeclaration (5,1),Main (7,1),Colon (7,5),Id "a" (8,5),Assign (8,7),IntValue 5 (8,9),SemiColon (8,10),While (9,5),LeftParenthesis (9,10),Id "a" (9,11),Less (9,13),IntValue 10 (9,15),RightParenthesis (9,17),Colon (9,18),Id "a" (10,9),Assign (10,11),Id "a" (10,13),Plus (10,15),IntValue 1 (10,17),SemiColon (10,18),EndWhile (11,5),SemiColon (11,14),EndMain (12,1)]